Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/AprilTagDetect.java

```diff
@@ -0,0 +1,122 @@
+package org.firstinspires.ftc.team6220_PowerPlay;
+
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.openftc.apriltag.AprilTagDetection;
+import org.openftc.easyopencv.OpenCvCamera;
+import org.openftc.easyopencv.OpenCvCameraFactory;
+import org.openftc.easyopencv.OpenCvCameraRotation;
+
+import java.util.ArrayList;
+import java.util.Locale;
+
+public abstract class AprilTagDetect extends BaseAutonomous {
+    OpenCvCamera camera;
+    AprilTagDetectionPipeline aprilTagDetectionPipeline;
+
+    // units are pixels
+    // calibration is for Logitech C920 webcam at 1920 x 1080
+    final double fx = 1387.853; // focal length x
+    final double fy = 1387.853; // focal length y
+    final double cx = 960; // camera principal point x
+    final double cy = 540; // camera principal point y
+
+    // units are meters
+    final double tagSize = 0.03429;
+
+    final int ID_TAG_OF_INTEREST_0 = 0; // tag 0 from the 36h11 family
+    final int ID_TAG_OF_INTEREST_1 = 1; // tag 1 from the 36h11 family
+    final int ID_TAG_OF_INTEREST_2 = 2; // tag 2 from the 36h11 family
+
+    AprilTagDetection tagOfInterest = null;
+
+    public int detectAprilTag() {
+
+        // initializes camera
+        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        camera = OpenCvCameraFactory.getInstance().createWebcam(hardwareMap.get(WebcamName.class, "Webcam 1"), cameraMonitorViewId);
+        aprilTagDetectionPipeline = new AprilTagDetectionPipeline(tagSize, fx, fy, cx, cy);
+
+        camera.setPipeline(aprilTagDetectionPipeline);
+        camera.openCameraDeviceAsync(new OpenCvCamera.AsyncCameraOpenListener() {
+            @Override
+            public void onOpened() {
+                camera.startStreaming(1920,1080, OpenCvCameraRotation.UPRIGHT);
+            }
+
+            @Override
+            public void onError(int errorCode) {
+
+            }
+        });
+
+        //initialize();
```

Comment:
I commented the method out because I was testing with just a control hub and not the robot so the program would crash because it didn't detect the motors in the initialize method. I will un-comment the code once I test the program on the robot.